src/transform/forward.rs

```diff
@@ -153,8 +153,31 @@ pub mod native {
       let row_coeffs = &mut buf[r * txfm_size_col..];
       txfm_func_row(row_coeffs);
       av1_round_shift_array(row_coeffs, txfm_size_col, -cfg.shift[2]);
-      for c in 0..txfm_size_col {
-        output[c * txfm_size_row + r] = T::cast_from(row_coeffs[c]);
+
+      // Store output in at most 32x32 chunks so that the first 32x32
+      // coefficients are stored first. When we don't have 64 rows, there is no
+      // change in order. With 64 rows, the chunks are in this order
+      //  - First 32 rows and first 32 cols
+      //  - Last 32 rows and first 32 cols
+      //  - First 32 rows and last 32 cols
+      //  - Last 32 rows and last 32 cols
+
+      // Output is grouped into 32x32 chunks so a stride of at most 32 is
+      // used for each chunk.
+      let output_stride = txfm_size_row.min(32);
+
+      // Split the first 32 rows from the last 32 rows
+      let output = &mut output
+        [(r >= 32) as usize * output_stride * txfm_size_col.min(32)..];
```

Comment:
Might be better to split the loop in two one for `< 32` and the other for `>= 32` instead of having the branch in the inner loop.